pkg/resource/domain/conditions.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package domain
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+)
+
+// getSyncedCondition returns the Condition in the resource's Conditions
+// collection that is of type ConditionTypeResourceSynced. If no such condition
+// is found, returns nil.
+//
+// TODO(jaypipes): Move to ACK code-gen templates.
+func getSyncedCondition(r *resource) *ackv1alpha1.Condition {
```

Comment:
these methods are already present in common-runtime. Why duplicate them?

https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/condition/condition.go

Reply:
> these methods are already present in common-runtime. Why duplicate them?
> 
> https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/condition/condition.go

Sorry, @vijtrip2 I copied from RDS controller :) I will submit separate PRs to bring these inline with common runtime for both opensearch and RDS.